main.gs

```diff
@@ -1,17 +1,32 @@
 function doPost(e) { // catches slack slash commands
   if (typeof e !== 'undefined') {
     
-    // extract message body
-    var par = e.parameter; 
+    // parse the event object
+    var slackEvent = new SlackEventWrapper();
```

Comment:
I wasn't sure the direct assignment from constructor would work since my parseEvent function doesn't return anything? Are you suggesting the following would work?

slack_wrapper.gs
```
constructor(e){
[...] // init all var to NULL
this.parseEvent(e);
}
```

main.gs
`var slackEvent = new SlackEventWrapper(e);`

Reply:
It should. 

You don't even need to init all variables to null. The main reason you would want to do that is if some values are not set on some code paths, if you try and access the with `this.thing` you would get an error.

Reply:
updated